acoustid.py

```diff
@@ -412,3 +417,54 @@ def get_submission_status(apikey, submission_id, timeout=None):
         'id': submission_id,
     }
     return _api_request(_get_submission_status_url(), params, timeout)
+
+
+def track_by_mbid(release_ids, disabled=False, timeout=None):
+    """Get AcoustID track id(s) corresponding to the given MusicBrainz
+    release id(s).
+    If ``release_ids`` is a str, a list of strs (possibly empty) is returned.
+    If ``release_ids`` is a list of strs, a dict mapping each str in that list
+    to a (possibly empty) list of strs is returned.
+    If ``disabled`` is True, those lists of str are instead pairs of lists of
+    strs, the first containing the enabled AcoustID track ids and the second the
+    disabled track ids."""
+
+    # Avoid isinstance(release_ids, list) in case the caller wants to pass some
+    # other sequence.  We let requests convert the sequence to a repeated param.
+    batch = not isinstance(release_ids, str)
+    params = {
+        'format': 'json',
+        'mbid': release_ids,
+        'disabled': '1' if disabled else '0',
+        'batch': '1' if batch else '0',
+        # this route doesn't require an API key
+    }
+
+    response = _api_request(_get_track_by_mbid_url(), params, timeout)
+    # Copied from submit, above.
+    if response.get('status') != 'ok':
+        try:
+            code = response['error']['code']
+            message = response['error']['message']
+        except KeyError:
+            raise WebServiceError("response: {0}".format(response))
+        raise WebServiceError("error {0}: {1}".format(code, message))
```

Comment:
Indeed, as you mentioned, maybe this can be factored out into a function to avoid duplication. The two options are for it to go into `_api_request` or into a new, separate function. The former would be more appropriate if we *only* ever use `_api_request` this way; the latter might be better if some calls to `_api_request` need to process stuff differently.